Mathlib/CategoryTheory/ChosenFiniteProducts.lean

```diff
@@ -549,6 +566,247 @@ end ChosenFiniteProductsComparison
 
 end ChosenFiniteProducts
 
+namespace ChosenFiniteCoproducts
+
+/-- The chosen initial object -/
+abbrev initialObj (C : Type u) [Category.{v} C] [ChosenFiniteCoproducts C] : C := initial.cocone.pt
+
+-- TODO: if `AddMonoidalCategory` is added, switch to use notation from there, analogously to how
+-- `ChosenFiniteProducts` uses notation from `MonoidalCategory`
+
+/-- Notation for the chosen initial object -/
+scoped notation "𝟘_" => initialObj
+
+variable {C : Type u} [Category.{v} C] [ChosenFiniteCoproducts C]
+
+/-- The chosen coproduct of two objects -/
+abbrev coprodObj (X Y : C) : C := (coproduct X Y).cocone.pt
+
+/-- Notation for the chosen coproduct of two objects -/
+scoped infixr:70 " ⊕ₒ " => coprodObj
+
+/--
+The unique map from the initial object.
+-/
+def fromZero (X : C) : 𝟘_ C ⟶ X :=
+  initial.isColimit.desc <| .mk _ <| .mk (fun x => x.as.elim) fun x => x.as.elim
+
+instance (X : C) : Unique (𝟘_ C ⟶ X) where
+  default := fromZero _
+  uniq _ := initial.isColimit.hom_ext fun ⟨j⟩ => j.elim
+
+/--
+This lemma follows from the preexisting `Unique` instance, but
+it is often convenient to use it directly as `apply fromZero_unique` forcing
+lean to do the necessary elaboration.
+-/
+lemma fromZero_unique {X : C} (f g : 𝟘_ _ ⟶ X) : f = g :=
+  Subsingleton.elim _ _
+
+/--
+Construct a morphism from the coproduct given its two components.
+-/
+def desc {T X Y : C} (f : X ⟶ T) (g : Y ⟶ T) : X ⊕ₒ Y ⟶ T :=
+  (coproduct X Y).isColimit.desc <| Limits.BinaryCofan.mk f g
+
+/--
+The first injection into the coproduct
+-/
+def inl (X Y : C) : X ⟶ X ⊕ₒ Y :=
+  letI F : Limits.BinaryCofan X Y := (coproduct X Y).cocone
+  F.inl
+
+/--
+The second injection into the coproduct
+-/
+def inr (X Y : C) : Y ⟶ X ⊕ₒ Y :=
+  letI F : Limits.BinaryCofan X Y := (coproduct X Y).cocone
+  F.inr
+
+@[simp, reassoc]
+lemma inl_desc {T X Y : C} (f : X ⟶ T) (g : Y ⟶ T) : inl X Y ≫ desc f g = f := by
+  simp [inl, desc]
+
+@[simp, reassoc]
```

Comment:
```suggestion
@[reassoc (attr := simp)]
```

Reply:
I now remember why these were not simp lemmas and reverted the change: `simp` can automatically prove the `reassoc` version